src/components/Theme/src/Theme.vue

```diff
@@ -20,21 +21,66 @@ function isVueComponent(value) {
 	return value && value.render;
 }
 
+// returns true if value is an array
+// of objects with id fields
+function isKeyedArray(value) {
+	return value?.[0]?.id;
+}
+
+function keyedArrayIntoObject(arrayValue) {
+	const objectValue = {};
+	for (const item of arrayValue) {
+		objectValue[item.id] = item;
+	}
+	return objectValue;
+}
+
+function objectIntoKeyedArray(objectValue) {
+	const arrayValue = [];
+	for (const [id, item] of Object.entries(objectValue)) {
+		item.id = id;
+		arrayValue.push(item);
+	}
+	return arrayValue;
+}
+
+function mergeKeyedArrays(value, mergeValue) {
+	const valueObject = keyedArrayIntoObject(value);
+	const mergeValueObject = keyedArrayIntoObject(mergeValue);
+	const mergedObject = merge(valueObject, mergeValueObject);
+	const mergedKeyedArray = objectIntoKeyedArray(mergedObject);
+	return mergedKeyedArray;
+}
+
 // recursively merge all objects
 // EXCEPT for Vue components
-function mergeStrategy(_value, mergeValue) {
+function mergeStrategy(value, mergeValue) {
 	if (isVueComponent(mergeValue)) {
 		return mergeValue;
 	}
 
+	if (isKeyedArray(value) && isKeyedArray(mergeValue)) {
+		return mergeKeyedArrays(value, mergeValue);
+	}
+
 	// returning undefined means "merge values recursively"
 	// https://lodash.com/docs/4.17.15#mergeWith
 	return undefined;
 }
 
 function resolveTheme(data, parentTheme, theme, profileId) {
 	mergeWith(data, parentTheme, theme, mergeStrategy);
-	mergeWith(data, find(data.profiles, { id: profileId }), mergeStrategy);
+	if (profileId) {
+		const foundProfile = find(data.profiles, { id: profileId });
```

Comment:
We can likely use the native `find` over the lodash one. https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore#_find